File: web-client/src/types/neighbourhood.ts
export type PolygonGeoJson = {
  type: 'Polygon';
  coordinates: number[][][];
};

export type NeighbourhoodDto = {
  id: string;
  name: string;
  description: string | null;
  boundary: PolygonGeoJson;
  createdAt: string;
  updatedAt: string;
};

File: desktop-client/src/main/java/com/vicinity/desktop/api/dto/Neighbourhood.java
package com.vicinity.desktop.api.dto;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.JsonNode;

@JsonIgnoreProperties(ignoreUnknown = true)
public record Neighbourhood(
        String id,
        String name,
        String description,
        JsonNode boundary,
        String createdAt,
        String updatedAt) {

    public String descriptionOrEmpty() {
        return description == null ? "" : description;
    }
}
